### PR TITLE
#102 several fixes for (dead) links in navigating HTML

### PR DESCRIPTION
**Pull request by justb4**
Appears this issue was not really due to #92. 
At least no dead links anymore and JSON content when accessed in browser. 

Question remains that on single Item the `Next` and `Prev` links should really walk through Collection but I suggest to do that in other issue, as this is quite tricky to achieve (as one does not know the prev/next id's in isolation of single Item).  

Hold on..one failing test now...

### CODE REVIEW

File: pygeoapi/flask_app.py [hunk 1]
```diff
@@ -166,3 +166,7 @@ def serve(ctx, debug=False):
 #    setup_logger(CONFIG['logging'])
     APP.run(debug=True, host=api_.config['server']['bind']['host'],
             port=api_.config['server']['bind']['port'])
+
+
+if __name__ == '__main__':  # run locally, for testing
```
Comment (tomkralidis) [inline pygeoapi/flask_app.py]: Do we need this?  `pygeoapi serve` accomplishes this already.

File: pygeoapi/api.py [hunk 2]
```diff
@@ -125,6 +125,10 @@ def root(self, headers, args):
         ]
 
         if format_ == 'html':  # render
+            for link in fcm['links']:
+                if 'json' in link['type']:
+                    link['href'] += '?f=json'
```
Comment (tomkralidis) [inline pygeoapi/api.py]: is this better written as `link['href'] = ''.join((link['href'], '?f=json'))`?  I'm not sure what the current verdict is around 'adding' strings in Python.

File: pygeoapi/flask_app.py (hunk 1 above)
Comment (justb4) [reply pygeoapi/flask_app.py]: Main reason for `main()` here is for (stepwise) debugging within an IDE like IntelliJ IDEA or WebStorm and probably PyCharm: needed easy entry point.

File: pygeoapi/api.py (hunk 2 above)
Comment (justb4) [reply pygeoapi/api.py]: From what I read both `+` and `join` can be used, with `join` being marginally faster with many strings to concatenate. For readablity `+` is preferred.

**Comment by justb4** [comment]
Checks failed since one test need change in `headers()` fixture to emulate real HTTP (`Werkzeug`) Headers structure with embedded `environ`. Will add. This would also allow for more advanced tests simulating many variations of client requests (think of proxied requests, using SCRIPT_NAME and returning full URLs within responses (i.s.o http://localhost:5000).

**Comment by justb4** [comment]
Also needed to fix the breadcrumbs relative URL paths: in cases where the page URL ends with `/` the breadcrumbs paths gave stale links.

Now all HTML links are based on Flask/Werkzeug `PATH_INFO`. This is more robust, as this also would work behind a proxy (when SCRIPT_NAME included).

**Pull request merged**